File: skysail.app.instagram.mocked/src/io/skysail/app/instagram/mocked/InstagramUser.java
package io.skysail.app.instagram.mocked;

import javax.persistence.Id;

import io.skysail.domain.Entity;
import io.skysail.domain.html.Field;
import io.skysail.server.codegen.annotations.GenerateResources;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
//@GenerateResources(application = "io.skysail.app.instagram.mocked.InstagramApiMockApplication")
public class InstagramUser implements Entity {

    @Id
    private String id;

    @Field
    private String username;

}

File: skysail.app.instagram/src/io/skysail/app/instagram/domain/InstagramUser.scala
package io.skysail.app.instagram.domain

import io.skysail.domain.html.Field
import io.skysail.domain.Entity
import io.skysail.domain.html.InputType

class InstagramUser(user: User) extends Entity {
  def getId(): String = user.data.id

  @Field 
  var username = "";
  def getUsername():String = user.data.username

  @Field(inputType = InputType.IMAGE)
  var picture = "";
  def getPicture():String = user.data.picture

}